optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,267 @@
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+                Can be either:
+                    - The model id of a pretrained model hosted inside a model repo on huggingface.co.
+                    - The path to a directory containing the model weights.
+            use_auth_token (`str` or `bool`):
+                The token to use as HTTP bearer authorization for remote files. Needed to load models from a private
+                repository.
+            revision (`str`, *optional*):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id.
+            cache_dir (`Union[str, Path]`, *optional*):
+                The path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            force_download (`bool`, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            file_name(`str`, *optional*):
+                The file name of the model to load. Overwrites the default file name and allows one to load the model
+                with a different name. This argument will be deprecated in next release.
+            local_files_only(`bool`, *optional*, defaults to `False`):
+                Whether or not to only look at local files (i.e., do not try to download the model).
+        """
+        if file_name is not None:
+            logger.warning("The argument of `file_name` will be deprecated in next release.")
+        else:
+            file_name = WEIGHTS_NAME
+        model_kwargs = {
+            "revision": revision,
+            "use_auth_token": use_auth_token,
+            "cache_dir": cache_dir,
+            "local_files_only": local_files_only,
+            "force_download": force_download,
+        }
+        if getattr(config, "torchscript", None):
+            # Load the model from local directory
+            if os.path.isdir(model_id):
+                file_name = os.path.join(model_id, file_name)
+                model_save_dir = model_id
+            # Download the model from the hub
+            else:
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    **model_kwargs,
+                )
+                model_save_dir = Path(model_cache_path).parent
+            model = cls.load_model(file_name)
+        else:
+            model_save_dir = None
+            task = cls.export_feature
+            if config.torch_dtype != "int8" and config.torch_dtype != torch.int8:
+                model = TasksManager.get_model_from_task(task, model_id, torch_dtype=torch_dtype, **model_kwargs)
+            else:
+                INCModel.TRANSFORMERS_AUTO_CLASS = cls.auto_model_class
```

Comment:
Shouldn't we use `_HEAD_TO_AUTOMODELS` instead ? https://github.com/huggingface/optimum-intel/blob/6a9b778115771c0972b7e66a3d0df88bc55d8ef0/optimum/intel/neural_compressor/utils.py#L34

Reply:
INCBaseModel didn't inherit from INCModel, so I need to set INCModel.TRANSFORMERS_AUTO_CLASS to AutoModelForCausalLM to create the right model, it seems no way to use _HEAD_TO_AUTOMODELS to get the right model.
If INCBaseModel inherits from INCModel, there are many issues in using the parent method.

Reply:
Instead of calling `INCModel.from_pretrained` it could make sense to call the right class directly such as `INCModelForSequenceClassification.from_pretrained`, `INCModelForQuestionAnswering.from_pretrained`, ... 
To get it you can use `_HEAD_TO_AUTOMODELS[cls.export_feature]`

Reply:
Now INCModelForCausalLM class have moved to modeling_decoder.py, and it didn't inherit from INCModel, so it is not able to use _HEAD_TO_AUTOMODELS[cls.export_feature].